ddev/tests/conftest.py

```diff
@@ -52,24 +76,29 @@ def isolation() -> Generator[Path, None, None]:
 
 
 @pytest.fixture(scope='session')
-def local_clone(isolation, local_repo) -> Generator[Path, None, None]:
-    cloned_repo = isolation / local_repo.name
+def local_clone(isolation, local_repo) -> Generator[ClonedRepo, None, None]:
+    cloned_repo_path = isolation / local_repo.name
+
+    PLATFORM.check_command_output(['git', 'clone', '--local', '--shared', str(local_repo), str(cloned_repo_path)])
+    with cloned_repo_path.as_cwd():
+        PLATFORM.check_command_output(['git', 'config', 'user.name', 'Foo Bar'])
+        PLATFORM.check_command_output(['git', 'config', 'user.email', 'foo@bar.baz'])
 
-    PLATFORM.check_command_output(['git', 'clone', '--local', '--shared', str(local_repo), str(cloned_repo)])
+    cloned_repo = ClonedRepo(cloned_repo_path, 'origin/master', 'ddev-testing')
+    cloned_repo.reset_branch()
 
     yield cloned_repo
 
 
 @pytest.fixture
-def repository(local_clone, config_file) -> Generator[Path, None, None]:
-    config_file.model.repos['core'] = str(local_clone)
+def repository(local_clone, config_file) -> Generator[ClonedRepo, None, None]:
```

Comment:
Let's say I want to test something with the -extras repo (or even marketplace). What do you think would be the best approach? Cloning the local -core repo and rename if to -extras to "fake" it, cloning the real -extras repo or building a fake -extras from scratch? Cloning the "real" repo might take some time.

Reply:
You mean running ddev tests against other repos? Interesting question. Maybe env var?

Reply:
Yep. Some of our commands behave differently depending on the repo, so we will need to provide some fixtures for that too. 

Not sure env vars would be the best option, will try to think about it :)

Reply:
Oh actually I have a way in next PR 🙂